parsons/etl/tofrom.py

```diff
@@ -675,10 +678,26 @@ def from_s3_csv(cls, bucket, key, aws_access_key_id=None, aws_secret_access_key=
         s3 = S3(aws_access_key_id, aws_secret_access_key)
         file_obj = s3.get_file(bucket, key)
 
-        if files.compression_type_for_path(key) == 'zip':
-            file_obj = files.zip_archive.unzip_archive(file_obj)
+        if from_manifest:
+            with open(file_obj) as fd:
+                manifest = json.load(fd)
 
-        return cls(petl.fromcsv(file_obj, **csvargs))
+            s3_files = [x["url"] for x in manifest["entries"]]
+
+        else:
+            s3_files = [file_obj]
+
+        tbls = []
+        for file in s3_files:
+            # TODO handle urls that end with '/', i.e. urls that point to "folders"
+            _, _, bucket_, key_ = files.split("/", 3)
```

Comment:
I think you want to split `file` here, not `files`.

Reply:
I haven't had a chance to test this out, but it looks like if `from_manifest_file` is `False`, then `file` will be a local file path (return value from `get_file`), where if `from_manifest_file` is `True`, then `file` will be the URL from the manifest file. I think you can move this `split` and `get_file` after you get the URLs from the manifest (line 685). That way `s3_files` will always be a list of local paths.